onnx/defs/nn/defs.cc

```diff
@@ -2487,4 +2487,194 @@ ONNX_OPERATOR_SET_SCHEMA(
         )ONNX"
         ));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+  // we need at least three inputs to have a shape for this inference.
+  if (!hasNInputShapes(ctx, 3)) {
```

Comment:
kernel shape is not being used in final_output_shape calculation so you can change this to 2 since only 2 inputs are required